### PR TITLE
Expand issuer property tests.

### PR DESCRIPTION
**Pull request by aljones15**

### CODE REVIEW

**Review by BigBlueHat — CHANGES_REQUESTED** [review]
This one's tricky. URLs are wonderfully varied and composable things, so I don't think most/any issuer would actually choke on the negative example here. A Verifier might...but only if/when it tried to dereference it.

File: tests/input/credential-issuer-object-id-not-url-fail.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "@context": [
+    "https://www.w3.org/ns/credentials/v2"
+  ],
+  "type": [
+    "VerifiableCredential"
+  ],
+  "validFrom": "2023-02-26T00:37:06Z",
+  "issuer": {
+    "id": "https ://not-a-url/issuer/foo"
```
[the remark this answers is not in the document]
Comment (aljones15) [reply tests/input/credential-issuer-object-id-not-url-fail.json:10]: I could replace this with a number or null, a number seems best. This has been tested on 2 implementations and this does actually cause all url parsers used by implementers to fail. However, I think a different type such as a number might get the issue across better?

File: tests/input/credential-issuer-object-id-not-url-fail.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "@context": [
+    "https://www.w3.org/ns/credentials/v2"
+  ],
+  "type": [
+    "VerifiableCredential"
+  ],
+  "validFrom": "2023-02-26T00:37:06Z",
```
[the remark this answers is not in the document]
Comment (aljones15) [reply tests/input/credential-issuer-object-id-not-url-fail.json:8]: not sure how that snuck in, I'll make a check for `validFrom` in the test data to ensure it's only on tests for that property.

[the remark this answers is not in the document]
Comment (aljones15) [reply tests/input/credential-issuer-object-id-not-url-fail.json]: I removed `validFrom` from all the VCs in this specific test: https://github.com/w3c/vc-data-model-2.0-test-suite/pull/50/commits/ecfa55febf5002f70d3d3069b042f640394c6274

https://github.com/w3c/vc-data-model-2.0-test-suite/pull/50/commits/3ce010952e0eeda6a0ef1d795c2d7c1cb55bd1bb

**Comment by dlongley** [comment]
Can we put a number or boolean in the field (non-string) instead? If we already have such a test, let's strike this, if not, let's just use a non-string to handle the test case here.

**Comment by aljones15** [comment]
> Can we put a number or boolean in the field (non-string) instead? If we already have such a test, let's strike this, if not, let's just use a non-string to handle the test case here.

we can change to a number or boolean @BigBlueHat thoughts here? I actually think this is a fine solution.

**Comment by BigBlueHat** [comment]
> we can change to a number or boolean @BigBlueHat thoughts here? I actually think this is a fine solution.

Fine by me.

**Review by JSAssassin — APPROVED** [review]
LGTM! Needs a changelog entry.

**Review by BigBlueHat — APPROVED** [review]

**Pull request merged**